driver-core/src/main/com/mongodb/internal/client/model/bulk/ConcreteClientUpdateManyModel.java

```diff
@@ -15,26 +15,32 @@
  */
 package com.mongodb.internal.client.model.bulk;
 
-import com.mongodb.client.model.bulk.ClientUpdateOptions;
+import com.mongodb.assertions.Assertions;
+import com.mongodb.client.model.bulk.ClientUpdateManyOptions;
 import com.mongodb.lang.Nullable;
 import org.bson.conversions.Bson;
 
 /**
  * This class is not part of the public API and may be removed or changed at any time.
  */
-public final class ConcreteClientUpdateManyModel extends AbstractClientUpdateModel implements ClientWriteModel {
+public final class ConcreteClientUpdateManyModel extends AbstractClientUpdateModel<ConcreteClientUpdateManyOptions> implements ClientWriteModel {
     public ConcreteClientUpdateManyModel(
             final Bson filter,
             @Nullable
             final Bson update,
             @Nullable
             final Iterable<? extends Bson> updatePipeline,
-            @Nullable final ClientUpdateOptions options) {
-        super(filter, update, updatePipeline, options);
+            @Nullable final ClientUpdateManyOptions options) {
+        super(filter, update, updatePipeline,
+                options == null ? ConcreteClientUpdateManyOptions.MUTABLE_EMPTY : (ConcreteClientUpdateManyOptions) options);
     }
 
     @Override
-    String getToStringDescription() {
-        return "ClientUpdateManyModel";
+    public String toString() {
+        return "ClientUpdateManyModel{"
+                + "filter=" + getFilter()
+                + ", update=" + (getUpdate().isPresent() ? getUpdate() : getUpdatePipeline().orElseThrow(Assertions::fail))
```

Comment:
With this code, we will add `Optional.toString` as the value of `update`. I propose

```suggestion
                + ", update=" + getUpdate().map(Object::toString).orElseGet(() ->
                        getUpdatePipeline().map(Object::toString).orElseThrow(Assertions::fail))
```

Formatting does not look good in the GitHub UI, but it's just fine in the code, when lines are not wrapped forcefully.